### PR TITLE
Force calender integration

### PR DESCRIPTION
**Pull request by ThushiniRanasinghe1998**

### CODE REVIEW

**Review by zrohacova — APPROVED** [review]

File: homeassistant/components/spotify/recommendation_handling.py [hunk 1]
```diff
@@ -146,13 +157,27 @@ def handling_date_recommendations(
...
+            search_string = HolidayDateMapper().search_string_date(
+                calendar_entity_ids, hass, user
+            )
+
+        except ValueError as e:
+            raise ValueError("Error generating search string: No user provided") from e
+        except AttributeError as e:
+            raise AttributeError(
+                "Error generating search string: The user does not have a country provided"
+            ) from e
```
Comment (amalev18) [inline homeassistant/components/spotify/recommendation_handling.py]: This try-except block can be completely removed! It was initially created in my branch, but after investigation it seems unnecessary. So it's removed, and should be removed here then aswell.

File: tests/components/spotify/test_recommendation_handling.py [hunk 2]
```diff
@@ -23,13 +25,77 @@ async def test_singleton_pattern(hass: HomeAssistant) -> None:
...
+    """Test that raised errors are handled correctly when propagated between modules."""
+    mock_holiday.return_value = NO_HOLIDAY
+    mock_season.return_value = "Summer"
+    mock_month.return_value = "July"
+    mock_weekday.return_value = "Sunday"
+
+    handler = RecommendationHandler()
+
+    with pytest.raises(ValueError):
+        handler._generate_date_search_string(hass, None)
```
Comment (amalev18) [inline tests/components/spotify/test_recommendation_handling.py]: The two above test fails for me? "homeassistant.exceptions.HomeAssistantError: No calendar integration found. Please set up a calendar integration." These are tests actually testing the call from _generate_date_search_string to the search string generator.

File: homeassistant/components/spotify/recommendation_handling.py (hunk 1 above)
Comment (ThushiniRanasinghe1998) [reply homeassistant/components/spotify/recommendation_handling.py]: Fixed this!

File: tests/components/spotify/test_recommendation_handling.py (hunk 2 above)
Comment (ThushiniRanasinghe1998) [reply tests/components/spotify/test_recommendation_handling.py]: I have corrected these test cases

**Comment by moa114** [comment]
This PR seems to contain changes that are already merged to main, maybe it can be solved by merging dev into this branch?

**Comment by ThushiniRanasinghe1998** [comment]
> This PR seems to contain changes that are already merged to main, maybe it can be solved by merging dev into this branch?

I have merged dev with this branch

**Review by amalev18 — APPROVED** [review]

**Pull request merged**